code/perception/src/lidar_distance.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+import rospy
+import ros_numpy
+import numpy as np
+import lidar_filter_utility
+from sensor_msgs.msg import PointCloud2, Range
+
+class LidarDistance():
+    """ See doc/06_perception/02_lidar_distance_utility.md on
+        how to configute this node
+    """
+
+    def callback(self, data):
+        """ Callback function, filters a PontCloud2 message
+            by restrictions defined in the launchfile.
+
+            Publishes a range message containing the farest and
+            the closest point of the filtered result. Additionally,
+            publishes the filtered result as PointCloud2
+            on the topic defined in the launchfile.
+
+        :param data: a PointCloud2
+        """
+        coordinates = ros_numpy.point_cloud2.pointcloud2_to_array(data)
+
+        # https://stackoverflow.com/questions/44295375/how-to-slice-a-numpy-ndarray-made-up-of-numpy-void-numbers
+        bit_mask = lidar_filter_utility.bounding_box(
+            coordinates,
+            max_x=rospy.get_param('~max_x', np.inf),
+            min_x=rospy.get_param('~min_x', np.inf),
+            max_y=rospy.get_param('~max_y', np.inf),
+            min_y=rospy.get_param('~min_y', np.inf),
+            max_z=rospy.get_param('~max_z', np.inf),
+            min_z=rospy.get_param('~min_z', np.inf),
+        )
+
+        # Filter coordinates based in generated bit_mask
+        coordinates = coordinates[bit_mask]
+
+        # Create pointcloud from manipulated data
+        coordinates_manipulated = ros_numpy \
+            .point_cloud2.array_to_pointcloud2(coordinates)
+        coordinates_manipulated.header = data.header
+
+        # Publish manipulated pointCloud2
+        self.pub_pointcloud.publish(coordinates_manipulated)
```

Comment:
Why we need to publish the manipulated point cloud?

Reply:
This is very helpful for debugging. In Rviz, the size of the points from this filtered point clouds is ver large. This way you can see blocking elements. E.g. here in the window:

<img width="384" alt="image" src="https://user-images.githubusercontent.com/10117610/225735524-91822f22-6f4f-4638-94aa-86c11dd53d59.png">

Reply:
Ah okay. Yes very nice feature